doc/whatsnew/2.14.rst

```diff
@@ -79,6 +79,12 @@ Other Changes
 
   Ref #5392
 
+* ``OptionsProviderMixIn`` has been replaced with ``ArgumentsProvider``. ``ArgumentsProvider`` is considered
+  private API and most methods that were public on ``OptionsProviderMixIn`` have now been deprecated and will
```

Comment:
Pretty much agree that we should clearly state that things are private API, we want to be able to change our internal freely.